bokehjs/src/lib/core/visuals.ts

```diff
@@ -154,6 +314,65 @@ export class Fill extends ContextProperties {
 
 Fill.prototype.attrs = Object.keys(mixins.fill())
 
+export class Hatch extends ContextProperties {
+
+  readonly hatch_color: p.ColorSpec
+  readonly hatch_alpha: p.NumberSpec
+  readonly hatch_scale: p.NumberSpec
+  readonly hatch_pattern: p.StringSpec
+  readonly hatch_weight: p.NumberSpec
+
+  cache_select(name: string, i: number): any {
+    let value: any
+    if (name == "pattern") {
+      this.cache_select("hatch_color", i)
+      this.cache_select("hatch_scale", i)
+      this.cache_select("hatch_pattern", i)
+      this.cache_select("hatch_weight", i)
+      const {hatch_color, hatch_scale, hatch_pattern, hatch_weight, hatch_extra} = this.cache
+      if (hatch_extra != null && hatch_extra.hasOwnProperty(hatch_pattern) ) {
+        const custom = hatch_extra[hatch_pattern]
+        this.cache.pattern = custom.execute(custom, {hatch_pattern, hatch_color, hatch_scale, hatch_weight})
+      } else {
+        this.cache.pattern = create_hatch_canvas(hatch_pattern, hatch_color, hatch_scale, hatch_weight)
+      }
+    } else
+      value = super.cache_select(name, i)
+
+    return value
+  }
+
+  set_value(ctx: Context2d): void {
+    const pattern = create_hatch_canvas(this.hatch_pattern.value(), this.hatch_color.value(), this.hatch_scale.value(), this.hatch_weight.value())
+    ctx.fillStyle = ctx.createPattern(pattern, 'repeat')!
+    ctx.globalAlpha = this.hatch_alpha.value()
+  }
+
+  get doit(): boolean {
+    return !(this.hatch_color.spec.value === null ||
+             this.hatch_alpha.spec.value == 0 ||
+             this.hatch_pattern.spec.value == " " ||
+             this.hatch_pattern.spec.value == "blank" ||
+             this.hatch_pattern.spec.value === null)
```

Comment:
I would remove `" "` and `"blank"`, because having three values indicating no action is an overkill. Note that comparing a value of a type that doesn't include `null` to `null` is unfortunately allowed in TypeScript (I'm working to disallow this on tslint level), so you have to add `null` to the respective type of `hatch_pattern` property, even if the compiler doesn't complain about lack thereof. It will, however, complain at use site (examples/tests), disallowing you to pass `null` as `hatch_pattern`'s value.

Reply:
Without them you can't have a blank pattern included with a hatch mapper